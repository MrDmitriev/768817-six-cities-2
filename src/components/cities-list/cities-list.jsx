import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {ActionCreator} from '../../reducer.js';

export class CitiesList extends Component {
  render() {
    const {cities, activeCity, setCity, setOffersList, offers} = this.props;

    const handleCityClick = (e) => {
      const city = e.target.id;
      setCity(city);
      setOffersList(city, offers);
    };
    return (
      <section className="locations container">
        <ul className="locations__list tabs__list">
          {cities.map((item) => {
            const isActive = activeCity === item;
            return (
              <li className="locations__item" key={item} onClick={handleCityClick}>
                <a className={`locations__item-link tabs__item ${isActive ? `tabs__item--active` : ``}`} href="#">
                  <span id={item}>{item}</span>
                </a>
              </li>
            );
          })}
        </ul>
      </section>
    );
  }

  componentDidMount() {
    const {offers, setDefaultCity} = this.props;
    setDefaultCity(offers);
  }

  componentDidUpdate(prevProps) {
    const {offers, activeCity, setOffersList} = this.props;
    if (prevProps.activeCity !== activeCity) {
      setOffersList(activeCity, offers);
    }
  }
}

CitiesList.propTypes = {
  cities: PropTypes.arrayOf(PropTypes.string),
  activeCity: PropTypes.string,
  offers: PropTypes.array,
  setCity: PropTypes.func,
  setOffersList: PropTypes.func,
  setDefaultCity: PropTypes.func,
};

export default connect(
    (state) => ({
      activeCity: state.city,
      state,
    }),
    (dispatch) => ({
      setCity: (city) => dispatch(ActionCreator.setCity(city)),
      setOffersList: (city, offers) => dispatch(ActionCreator.setOffersList(city, offers)),
      setDefaultCity: (offers) => dispatch(ActionCreator.setDefaultCity(offers)),
    })
)(CitiesList);
