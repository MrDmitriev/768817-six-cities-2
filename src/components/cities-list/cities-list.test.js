import React from 'react';
import renderer from 'react-test-renderer';
import {CitiesList} from './cities-list.jsx';

it(`should match snapshot`, () => {
  const setCity = jest.fn();
  const setOffersList = jest.fn();
  const setDefaultCity = jest.fn();
  const props = {
    cities: [`AAA`],
    activeCity: `AAA`,
    offers: [{
      name: `AAA`,
      price: 120,
      type: `AAA`,
      src: `AAA`,
    }],
    setCity,
    setOffersList,
    setDefaultCity,
  };

  const wrapper = renderer.create(<CitiesList {...props} />).toJSON();

  expect(wrapper).toMatchSnapshot();
});
