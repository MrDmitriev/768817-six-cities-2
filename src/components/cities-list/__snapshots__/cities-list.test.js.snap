// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        href="#"
      >
        <span
          id="AAA"
        >
          AAA
        </span>
      </a>
    </li>
  </ul>
</section>
`;
